app/routes/task_routes.py

```diff
@@ -0,0 +1,90 @@
+from app import db
+from flask import Blueprint, jsonify, make_response, request
+from app.models.task import Task
+import datetime
+from .helpers import get_task_from_id, send_completion_slack_message
+from sqlalchemy.exc import DataError
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+
+# Task CRUD routes
+@tasks_bp.route("", methods=["GET"], strict_slashes=False)
+def get_all_tasks():
+    sort_query = request.args.get("sort")
+    if sort_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif sort_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all()
+    tasks_response = [task.to_dict() for task in tasks]
+    return make_response(jsonify(tasks_response), 200)
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_new_task():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body or \
+        "completed_at" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)
+    new_task = Task(
+        title = request_body["title"],
+        description = request_body["description"],
+        completed_at = request_body["completed_at"]
+    )
```

Comment:
pseudo code on how to return specific error messages

error_response = {"error": []}
error400 = False

if "title" not in request_body:
   error_response["error"].append("missing title")
   error400 = True
if "description" not in request_body:
   error_response["error"].append("missing description")
   error400 = True